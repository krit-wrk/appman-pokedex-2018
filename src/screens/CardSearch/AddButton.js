import React from "react";
import { css } from "aphrodite";
import style from "./style";

export const AddButton = ({ onClick }) => (
	<div className={css(style.addButton)} onClick={onClick}>
		Add
	</div>
);
