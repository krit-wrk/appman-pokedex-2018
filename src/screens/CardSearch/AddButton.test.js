import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { AddButton } from "./AddButton";
import { StyleSheetTestUtils } from "aphrodite";
describe("AddButton", () => {
	beforeEach(() => {
		StyleSheetTestUtils.suppressStyleInjection();
	});
	afterEach(() => {
		StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
	});
	it("renders without crashing", () => {
		const fn = jest.fn();
		const { container, getByText } = render(<AddButton onClick={fn} />);
		expect(container).toMatchSnapshot();
		fireEvent.click(getByText("Add"));
		expect(fn).toHaveBeenCalled();
	});
});
