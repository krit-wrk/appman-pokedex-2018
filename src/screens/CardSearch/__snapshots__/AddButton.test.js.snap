// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton renders without crashing 1`] = `
<div>
  <div
    class="addButton_5gyw2"
  >
    Add
  </div>
</div>
`;
